### PR TITLE
feat(abstract-utxo): add psbt support backup recovery

### PR DESCRIPTION
**Pull request by saravanan7mani**

### CODE REVIEW

File: modules/abstract-utxo/src/recovery/backupKeyRecovery.ts [hunk 1]
```diff
@@ -62,16 +55,17 @@ export interface FormattedOfflineVaultTxInfo {
  * @param txHex
  * @returns {{txHex: *, txInfo: {unspents: *}, feeInfo: {}, coin: void}}
  */
-function formatForOfflineVault<TNumber extends number | bigint = number>(
```
Comment (davidkaplanbitgo) [inline modules/abstract-utxo/src/recovery/backupKeyRecovery.ts:65]: why would you take away `TNumber` here?

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 2]
```diff
@@ -474,7 +474,11 @@ export function isTransactionWithKeyPathSpendInput(
  * extendedPubkey.
  */
 export function addXpubsToPsbt(psbt: UtxoPsbt, rootWalletKeys: RootWalletKeys): void {
-  const xPubs = rootWalletKeys.triple.map(
+  const safeRootWalletKeys = new RootWalletKeys(
+    rootWalletKeys.triple.map((bip32) => bip32.neutered()) as Triple<BIP32Interface>,
+    rootWalletKeys.derivationPrefixes
+  );
+  const xPubs = safeRootWalletKeys.triple.map(
```
Comment (davidkaplanbitgo) [inline modules/utxo-lib/src/bitgo/wallet/Psbt.ts:481]: nice

**Review by davidkaplanbitgo — COMMENTED** [review]
were all of the fixtures changed because you changed the mocking? Was there a reason why that was needed?

File: modules/abstract-utxo/src/recovery/backupKeyRecovery.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/abstract-utxo/src/recovery/backupKeyRecovery.ts:65]: As per new change, `OfflineVaultTxInfo` uses valueString to carry large values. value is always `number`. So no need to for Generics.

**Comment by saravanan7mani** [comment]
> were all of the fixtures changed because you changed the mocking? Was there a reason why that was needed?

Primarily what you mentioned (prevTx id change changes txHex) and also few cases returns PSBT hex instread transaction hex, new response fields `recoveryAmountString`.

File: modules/abstract-utxo/src/recovery/backupKeyRecovery.ts [hunk 3]
```diff
@@ -276,6 +283,7 @@ export async function backupKeyRecovery<TNumber extends number | bigint = number
 
   const isKrsRecovery = getIsKrsRecovery(params);
   const isUnsignedSweep = getIsUnsignedSweep(params);
+  const respondWithTx = isUnsignedSweep || !isKrsRecovery || params.krsProvider === 'keyternal';
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/recovery/backupKeyRecovery.ts:286]:
```suggestion
  const responseFormat = (isUnsignedSweep || !isKrsRecovery || params.krsProvider === 'keyternal') ? 'legacy' : 'psbt';
```

File: modules/abstract-utxo/src/recovery/backupKeyRecovery.ts [hunk 4]
```diff
@@ -324,50 +334,39 @@ export async function backupKeyRecovery<TNumber extends number | bigint = number
   // segwit overhead size and p2sh inputs for the same reason.
   const outputSize = (isKrsRecovery ? 2 : 1) * VirtualSizes.txP2wshOutputSize;
   const approximateSize = VirtualSizes.txSegOverheadVSize + outputSize + VirtualSizes.txP2shInputSize * unspents.length;
-  const approximateFee: TNumber = utxolib.bitgo.toTNumber<TNumber>(approximateSize * feePerByte, coin.amountType);
+  const approximateFee = BigInt(approximateSize * feePerByte);
 
-  // Construct a transaction
-  txInfo.inputs = unspents;
+  txInfo.inputs = respondWithTx
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/recovery/backupKeyRecovery.ts:339]:
```suggestion
  txInfo.inputs = responseFormat === 'legacy' ?
```

File: modules/bitgo/test/v2/unit/coins/utxo/recovery/backupKeyRecovery.ts [hunk 5]
```diff
@@ -203,32 +237,38 @@ function run(
 
     if (params.hasUserSignature && params.hasBackupSignature) {
       it('has no placeholder signatures', function () {
```
Comment (OttoAllmendinger) [inline modules/bitgo/test/v2/unit/coins/utxo/recovery/backupKeyRecovery.ts:239]:
```suggestion
      it('has no placeholder signatures', function (this: mocha.Context) {
```
then you can get rid of the `@ts-expect-error` further down

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts:481]: maybe we need `RootWalletPublicKeys`

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
nits mostly

**Review by OttoAllmendinger — APPROVED** [review]

**Review by zahin-mohammad — APPROVED** [review]

**Pull request merged**